radicle-node/src/worker/fetch.rs

```diff
@@ -398,6 +398,16 @@ impl<'a> StagingPhaseFinal<'a> {
         let head = production.set_identity_head()?;
         log::debug!(target: "worker", "'refs/rad/id' for {} set to {head}", production.id);
 
+        #[cfg(test)]
```

Comment:
in this case, we are not skipping this code while `cargo install --locked --path ./radicle-node`? I should run the node with cargo if I want to run this?

Is this the intention?

Reply:
Sorry, I'm not sure what you're asking.

The intention is that `debug_assert_matches` is only available with the test feature enabled (or at least it should be). That's fine because we only want this to run when testing and debugging.

Reply:
>Sorry, I'm not sure what you're asking.

I'm concerned that this code won't be included if we install the node with `cargo install --locked --path ./radicle-node`. 

In other words, it would be beneficial to have this code not only in the test but also in the production environment to provide more useful information if something goes wrong, no?

Reply:
It's intentionally only for debug, since it's expensive to run, and redundant if the code is correct.